ledger/src/lib.rs

```diff
@@ -67,14 +67,18 @@ use aleo_std::prelude::{finish, lap, timer};
 use anyhow::Result;
 use core::ops::Range;
 use indexmap::IndexMap;
+use lru::LruCache;
 use parking_lot::RwLock;
 use rand::{prelude::IteratorRandom, rngs::OsRng};
-use std::{borrow::Cow, sync::Arc};
+use std::{borrow::Cow, num::NonZeroUsize, sync::Arc};
 use time::OffsetDateTime;
 
 #[cfg(not(feature = "serial"))]
 use rayon::prelude::*;
 
+/// The number of transactions to keep cached in order to not perform redundant checks.
+const NUM_CACHED_TRANSACTIONS: usize = 128;
```

Comment:
How was the number of cached transactions determined?

Reply:
it was a placeholder; in any case, as long as we only need to keep the transaction IDs, it could be as large as the maximum number of transactions per block.